src/care/LoopFuser.h

```diff
@@ -814,10 +1118,41 @@ void LoopFuser::registerFree(care::host_device_ptr<T> & array) {
       0, 1, __fusible_scan_pos__, \
       [=] FUSIBLE_DEVICE(int, bool, int, int, int)->bool { return true; }, \
       [=] FUSIBLE_DEVICE(int, bool, int, int, int)->int {
+
+#define FUSIBLE_LOOP_PHASE(INDEX, START, END, PRIORITY) { \
+   if (END > START) { \
```

Comment:
You use very similar code multiple times. Is there a way to reuse the same code without obscuring everything? I think that might make it less error prone in the future if we adjust these macros.

Reply:
just pushed some changes that take a pass at this

Reply:
That looks better! Thanks!